libs/dal/src/repositories/channel-endpoint/channel-endpoint.schema.ts

```diff
@@ -0,0 +1,68 @@
+import { ChannelEndpointRoutingTypeEnum } from '@novu/shared';
+import mongoose, { Schema } from 'mongoose';
+import { schemaOptions } from '../schema-default.options';
+import { ChannelEndpointDBModel } from './channel-endpoint.entity';
+
+const routingSchema = new Schema(
+  {
+    type: {
```

Comment:
wonder if `providerId` is a better name? As those will be mostly provider specific? Or you thought of other usecase here?